src/check/canonicalize.zig

```diff
@@ -963,6 +1207,52 @@ pub fn canonicalize_expr(
             return expr_idx;
         },
         .field_access => |field_access| {
+            // Check if this is a module-qualified lookup (like Json.utf8)
+            const left_expr = self.parse_ir.store.getExpr(field_access.left);
+            if (left_expr == .ident) {
+                const left_ident = left_expr.ident;
+                if (self.parse_ir.tokens.resolveIdentifier(left_ident.token)) |module_alias| {
```

Comment:
Not a blocker, but this code is really deeply nested and a little hard to follow.  Since there is no error reporting happening in an `else` block, maybe we could just do something like

```zig
if (left_expr != .ident) {
    break;
}
const left_ident = left_expr.ident;
const module_alias = self.parse_ir.tokens.resolveIdentifier(left_ident.token) orelse break;
const module_name = self.scopeLookupModule(module_alias) orelse break;
const right_expr  = self.parse_ir.store.getExpr(field_access.right);
if (right_expr != .ident) {
    break;
}
const right_ident = right_expr.ident;
const field_name = self.parse_ir.tokens.resolveIdentifier(right_ident.token) orelse break;
....
```

Reply:
Significantly refactored in commit 4486b2466e637c6cbe6e5e762c13ffdd80cc7b6f, basically pulled out into helpers and added plenty of doc comments. It's much easier to follow now.